File: src/main/java/moon/scala/RemoveElement.scala
package moon.scala

class RemoveElement {
  def removeElement(nums:Array[Int],`val`:Int):Int={
    if(nums.length==0){
      return 0
    }
    var i=0
    for(j<-nums.indices){
      if(nums(j)!=`val`){
        nums(i)=nums(j)
        i+=1
      }
    }
    i
  }
}

File: data-structure/src/main/java/array/RemoveElement.java
package array;

public class RemoveElement {
    public int solve(int[] nums, int val) {
        int length=0;
        for(int i=0;i<nums.length;i++){
            if(nums[i]==val){
                continue;
            }
            nums[length++]=nums[i];
        }
        return length;
    }
}
